pkg/topology/kademlia/internal/metrics/metrics.go

```diff
@@ -18,14 +18,50 @@ import (
 	"github.com/syndtr/goleveldb/leveldb"
 )
 
+const ewmaSmoothing = 0.1
+
 // PeerConnectionDirection represents peer connection direction.
 type PeerConnectionDirection string
 
 const (
 	PeerConnectionDirectionInbound  PeerConnectionDirection = "inbound"
 	PeerConnectionDirectionOutbound PeerConnectionDirection = "outbound"
+)
+
+// PeerReachabilityStatus represents the peer reachability status.
+type PeerReachabilityStatus int
+
+// String implements the fmt.Stringer interface.
+func (rs PeerReachabilityStatus) String() string {
+	str := [...]string{reachabilityUnknown, reachabilityPublic, reachabilityPrivate}
+	if rs < 0 || int(rs) >= len(str) {
+		return "(unrecognized)"
+	}
+	return str[rs]
+}
 
-	ewmaSmoothing = 0.1
+// ParseReachabilityStatus tries to parse reachability status from the given string.
+func ParseReachabilityStatus(s string) (PeerReachabilityStatus, error) {
+	switch s {
+	case reachabilityUnknown:
+		return PeerReachabilityStatusUnknown, nil
+	case reachabilityPublic:
+		return PeerReachabilityStatusPublic, nil
+	case reachabilityPrivate:
+		return PeerReachabilityStatusPrivate, nil
+	}
+	return -1, fmt.Errorf("unrecognized reachability status: %q", s)
+}
+
+const (
+	PeerReachabilityStatusUnknown PeerReachabilityStatus = 0 // Mirrors the network.ReachabilityUnknown.
```

Comment:
have a look at https://github.com/ethersphere/bee/blob/master/pkg/storage/store.go#L26 as an example of a bit more simplified use case (using `iota` keyword). consider having the method on the status type but perhaps named `UnmarshalText`. also, consider defining the type inside the `p2p` package (where all the p2p contracts are defined) or in the `topology` package